File: global/src/main/java/ms2709/global/UseCase.java
package ms2709.global;

import org.springframework.core.annotation.AliasFor;
import org.springframework.stereotype.Component;

import java.lang.annotation.*;
/**
 * UseCase의 역할을 나타내기 위한 어노테이션
 *
 * @author 심문섭
 * @version 1.0
 * @class UseCase
 * @since 2024-05-02 오전 8:26
 */
@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Component
public @interface UseCase {
    /**
     * The value may indicate a suggestion for a logical component name,
     * to be turned into a Spring bean in case of an autodetected component.
     * @return the suggested component name, if any (or empty String otherwise)
     */
    @AliasFor(annotation = Component.class)
    String value() default "";

}

File: member-service/src/main/kotlin/ms2709/member/global/UseCase.kt
package ms2709.member.global

import org.springframework.stereotype.Component
import java.lang.annotation.Documented

/**
 *
 * UseCase의 역할을 나타내기 위한 어노테이션
 *
 * @class UseCase
 * @author 심문섭
 * @version 1.0
 * @since 2024-04-29 8:58 AM
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@Component
annotation class UseCase(val value: String = "") {

}